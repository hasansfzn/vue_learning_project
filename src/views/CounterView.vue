<script setup>
  import { ref } from "vue";
  import { counter } from "@/stores/counter";

  // const counter = ref(0);
  // const disabledButton = ref(false);

  const msg = ref("");
  const inputMsg = ref("");

  // const increase = () => {
  //   counter.value++;
  //   if (disabledButton.value) disabledButton.value = false;
  // };
  // const decrease = () => {
  //   if (counter.value > 0) counter.value--;
  //   else disabledButton.value = true;
  // };
</script>

<template>
  <div class="m-2">
    <h2 class="p-2 text-green-300 shadow-md font-semibold text-xl m-2">
      Counter is : {{ counter.count }}
    </h2>
    <div class="flex items-center justify-between gap-4">
      <button
        id="increment"
        class="p-2 btn rounded-2 border shadow shadow-gray-200"
        @click="counter.increment"
      >
        Increment
      </button>
      <button
        id="decrement"
        class="p-2 btn rounded-2 border shadow shadow-gray-200"
        :class="{ 'cursor-not-allowed': counter.disabled }"
        @click="counter.decrement"
        :disabled="counter.disabled"
      >
        Decrement
      </button>
    </div>
  </div>
  <div class="mt-5 pt-3">
    <h2
      class="mt-4 font-bold"
      id="message"
    >
      {{ msg }}
    </h2>
    <div class="flex">
      <input
        type="text"
        v-model="inputMsg"
        class="mt-2"
        placeholder="Enter a Message"
      />
      <button
        id="inputMessage"
        @click="msg = inputMsg"
        class="btn shadow shadow-gray-200 ms-2"
      >
        Save
      </button>
    </div>
  </div>
</template>

<style></style>
