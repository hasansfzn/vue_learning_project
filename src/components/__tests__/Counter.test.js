import { describe, it, expect } from "vitest";
import CounterView from "../../views/CounterView.vue";
import { mount } from "@vue/test-utils";

describe("Counter", () => {
  const wrapper = mount(CounterView);
  it("Should Render Correctly", () => {
    //as the next line will use in every it, we moves it to the upper level,
    // const wrapper = mount(CounterView);

    //instead of toMatchSnapShot(), we are using toContain
    // expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.html()).toContain("Counter is : 0");
  });

  it("Should Increment", async () => {
    await wrapper.find("#increment").trigger("click");
    expect(wrapper.html()).toContain("Counter is : 1");
  });

  it("Should Increment", async () => {
    await wrapper.find("#decrement").trigger("click");
    expect(wrapper.html()).toContain("Counter is : 0");
  });
});
