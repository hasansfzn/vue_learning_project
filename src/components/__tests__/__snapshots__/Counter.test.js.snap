// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > Should Increment 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="p-2 btn rounded-2 border shadow shadow-gray-200"
    id="increment"
  >
     Increment 
  </button>,
}
`;

exports[`Counter > Should Increment 2`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="p-2 btn rounded-2 border shadow shadow-gray-200"
    id="increment"
  >
     Increment 
  </button>,
}
`;

exports[`Counter > Should Render Correctly 1`] = `
"<div class=\\"m-2\\">
  <h2 class=\\"p-2 text-green-300 shadow-md font-semibold text-xl m-2\\"> Counter is : 0</h2>
  <div class=\\"flex items-center justify-between gap-4\\"><button class=\\"p-2 btn btn-primary rounded-2 border shadow shadow-gray-200\\"> Increment </button><button class=\\"p-2 btn btn-danger rounded-2 border shadow shadow-gray-200\\"> Decrement </button></div>
</div>"
`;
